core/services/relay/evm/mercury/transmitter.go

```diff
@@ -275,7 +277,7 @@ func (s *server) runQueueLoop(stopCh services.StopChan, wg *sync.WaitGroup, feed
 	}
 }
 
-func NewTransmitter(lggr logger.Logger, cfg TransmitterConfig, clients map[string]wsrpc.Client, fromAccount ed25519.PublicKey, jobID int32, feedID [32]byte, orm ORM, codec TransmitterReportDecoder) *mercuryTransmitter {
+func NewTransmitter(lggr logger.Logger, cfg TransmitterConfig, clients map[string]wsrpc.Client, fromAccount ed25519.PublicKey, jobID int32, feedID [32]byte, orm ORM, codec TransmitterReportDecoder, triggerCapability *triggers.MercuryTriggerService) *mercuryTransmitter {
```

Comment:
so many args is yellow flag. i'd prefer to see them consolidated into an opt or config struct and constructor handle/validate as needed. passing `nil` in constructors is a signal that we are really passing options

Reply:
I'd rather leave that to code owners here, I'm only adding the trigger service and don't fully understand the consequences of refactoring.